apm-lambda-extension/extension/http_server_test.go

```diff
@@ -137,3 +143,116 @@ func Test_getDecompressedBytesFromRequestEmptyBody(t *testing.T) {
 		t.Fail()
 	}
 }
+
```

Comment:
Is there a reason you didn't use the [httptest](https://pkg.go.dev/net/http/httptest) package here?
It might be a lot easier to write this test using it. 

Here is an example in the apm server repo: https://github.com/elastic/apm-server/blob/master/sourcemap/fleet_store_test.go

You could create a mock server using httptest for the apm server and write a handler that asserts directly on the request values. I know it would mean writing this all over again but I think it's the standard way to test http servers.

Let me know if I'm missing something and you'd like to stick with this approach.